libs/dyn/yamlsaver/utils.go

```diff
@@ -0,0 +1,46 @@
+package yamlsaver
+
+import (
+	"slices"
+
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/databricks/cli/libs/dyn/convert"
+)
+
+var skipFields = []string{"format"}
+
+// Converts a struct to map. Skips any nil fields.
+// It uses `skipFields` to skip unnecessary fields.
+// Uses `order` to define the order of keys in resulting outout
+func ConvertToMapValue(strct any, order *Order, dst map[string]dyn.Value) (dyn.Value, error) {
+	ref := dyn.NilValue
+	mv, err := convert.FromTyped(strct, ref)
+	if err != nil {
+		return dyn.NilValue, err
+	}
+
+	return skipAndOrder(mv, order, dst)
+}
+
+func skipAndOrder(mv dyn.Value, order *Order, dst map[string]dyn.Value) (dyn.Value, error) {
+	for k, v := range mv.MustMap() {
```

Comment:
This can panic if not a map. Maybe add a runtime check to `ConvertToMapValue` that it actually returns a map value (you could pass it a string and it wouldn't complain).